services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/service/implement/SubnetServiceImp.java

```diff
@@ -319,6 +318,21 @@ public boolean verifyCidrBlock(String cidr) throws FallbackException {
 
     }
 
+    @Override
+    @DurationStatistics
+    public Integer getUsedIpByRangeId(String rangeId) {
+        String ipManagerServiceUrl = ipUrl + "range" + "/" + rangeId;
+        IpAddrRangeRequest ipAddrRangeRequest = restTemplate.getForObject(ipManagerServiceUrl, IpAddrRangeRequest.class);
+        if (ipAddrRangeRequest == null) {
+            logger.info("can not find ipAddrRangeRequest by range id");
+            return null;
+        }
+
+        Integer usedIps = Integer.parseInt(String.valueOf(ipAddrRangeRequest.getUsedIps()));
```

Comment:
I think we need to either add validation before the conversion or add exception handling.

Reply:
okay